metrics_utility/automation_controller_billing/extract/extractor_common.py

```diff
@@ -87,29 +87,53 @@ def process_tarballs(self, path, temp_dir):
-    else:
-        indirect_nodes = pd.DataFrame([{}])
-
-    if os.path.exists(os.path.join(temp_dir, 'main_jobevent.csv')):
-        main_jobevent = pd.read_csv(os.path.join(temp_dir, 'main_jobevent.csv'))
-    else:
-        main_jobevent = pd.DataFrame([{}])
-
-    if os.path.exists(os.path.join(temp_dir, 'main_host.csv')):
-        main_host = pd.read_csv(os.path.join(temp_dir, 'main_host.csv'))
-    else:
-        main_host = pd.DataFrame([{}])
-
-
-    return {'main_jobevent': main_jobevent,
-            'main_host': main_host,
-            'job_host_summary': job_host_summary,
-            'indirect_nodes' : indirect_nodes,
-            'config': config}
+    empty_dataframe = pd.DataFrame([{}])
+    needed_data = {
+        'config': config,
+        'job_host_summary':empty_dataframe,
+        'indirect_nodes':empty_dataframe,
+        'main_jobevent':empty_dataframe,
+        'main_host':empty_dataframe
+    }
+    if required_sheets(self,['ccsp_summary','managed_nodes','indirectly_managed_nodes','inventory_scope',
+                             'usage_by_organizations','managed_nodes_by_organizations']):
+        needed_data['job_host_summary'] = build_data_batch(temp_dir,'job_host_summary')
+
+    if required_sheets(self,['managed_nodes', 'indirectly_managed_nodes', 'usage_by_organizations']):
+        needed_data['indirect_nodes']= build_data_batch(temp_dir,'main_indirectmanagednodeaudit')
+
+    if required_sheets(self,['usage_by_collections','usage_by_roles','usage_by_modules','usage_by_organizations']):
+        needed_data['main_jobevent'] = build_data_batch(temp_dir,'main_jobevent')
+
+    if required_sheets(self,['jobs','managed_nodes_by_organizations','managed_nodes','inventory_scope',
+                             'usage_by_collections,usage_by_roles,usage_by_modules','managed_nodes']):
+        needed_data['main_host']= build_data_batch(temp_dir,'main_host')
+
+    return needed_data
+
+
+def build_data_batch(temp_dir, file_name):
+      """
+      Builds the report with only the necessary sheets.
+      """
+
+      if os.path.exists(os.path.join(temp_dir, f'{file_name}.csv')):
+        return pd.read_csv(os.path.join(temp_dir, f'{file_name}.csv'))
+      else:
+        return pd.DataFrame([{}])
+
+
+def required_sheets(self, sheets_required):
+    """
+    Checks for METRICS_UTILITY_OPTIONAL_CCSP_REPORT_SHEETS values and return those values.
+    If none found, give it the default CCSP report sheet options.
+    Returns a boolean so we know which sheets to provide in the report.
+
+    """
+    sheet_options = os.getenv('METRICS_UTILITY_OPTIONAL_CCSP_REPORT_SHEETS','ccsp_summary,managed_nodes,\
+                              indirectly_managed_nodes,inventory_scope,usage_by_organizations,\
+                              managed_nodes_by_organizations').split(',')
+    if sheet_options is None: ## We might not need this..split(',')
+        sheet_options = self.optional_report_sheets()
```

Comment:
```suggestion
    sheet_options = self.optional_report_sheets()
```

the first part does exactly what the function does, so let's just keep the function call?